packages/backend/src/ee/services/AiAgentService.ts

```diff
@@ -1056,6 +1057,58 @@ export class AiAgentService {
         });
     }
 
+    private async storeThreadContextMessages(
+        threadUuid: string,
+        threadMessages: Array<
+            Required<Pick<MessageElement, 'text' | 'user' | 'ts'>>
+        >,
+        slackChannelId: string,
+        fallbackUserUuid: string,
+    ): Promise<void> {
+        if (threadMessages.length === 0) return;
+
+        // Get timestamps to check for existing prompts
+        const timestamps = threadMessages.map((msg) => msg.ts);
+        const existingTimestamps =
+            await this.aiAgentModel.existsSlackPromptsByChannelAndTimestamps(
+                slackChannelId,
+                timestamps,
+            );
+
+        // Filter out messages that already exist
+        const newMessages = threadMessages.filter(
+            (msg) => !existingTimestamps.includes(msg.ts),
+        );
+
+        if (newMessages.length === 0) return;
+
+        // Convert Slack timestamp to Date (Slack ts is Unix timestamp with microseconds)
+        const convertSlackTsToDate = (ts: string): Date =>
+            new Date(parseFloat(ts) * 1000); // Convert to milliseconds
+
+        // Prepare data for bulk insert
+        const promptsData = newMessages
+            .map((msg) => ({
+                createdByUserUuid: fallbackUserUuid, // TODO: use the user uuid from the message
```

Comment:
We have a method in slack client that gets actual slack users info, you can do it like this:

https://github.com/lightdash/lightdash/blob/ac1cd0fafa86374e0a176a5b1e3a7adea31f5316/packages/backend/src/ee/services/AiAgentService.ts#L355-L366

Reply:
Looking into this now